### PR TITLE
feat(kernel-test): add tests for manual vat termination, malformed KRef validation, and revoked object handling

### PR DESCRIPTION
**Pull request by sirtimid**
Ref: #550 

Add comprehensive integration tests for kernel edge cases and new functionality that weren't previously covered:

**Manual Vat Termination & Cleanup**
- Tests `kernel.terminateVat()` functionality and subsequent garbage collection
- Verifies proper cleanup of terminated vat objects from kernel store
- Ensures kernel remains stable after manual vat termination

**Malformed KRef Validation** 
- Tests kernel resilience to invalid KRef inputs (empty strings, malformed references)
- Verifies appropriate error handling without kernel crashes
- Ensures kernel remains operational after malformed requests

**Revoked Object Handling**
- Tests message delivery to revoked objects returns proper error responses
- Verifies promise rejection handling for revoked objects
- Tests bulk revocation scenarios with multiple objects
- Ensures kernel stability during revocation operations

### CODE REVIEW

File: packages/kernel-test/src/syscall-validation.test.ts [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+      true,
+      logger.logger.subLogger({ tags: ['test'] }),
+    );
+    // Test that truly malformed krefs cause kernel to throw (correct behavior)
+    await expect(
+      kernel.queueMessage('invalid-kref' as KRef, 'testMethod', ['test']),
+    ).rejects.toThrow('invalid reference context');
+    await expect(
+      kernel.queueMessage('' as KRef, 'testMethod', ['test']),
+    ).rejects.toThrow('incrementRefCount called with empty kref');
```
[the remark this answers is not in the document]
Comment (grypez) [reply packages/kernel-test/src/syscall-validation.test.ts]: [(and the integrity check performed w.r.t. each offending operation)](https://github.com/MetaMask/ocap-kernel/commit/a3d7b95e73024fef1ac285a6218d897960380187#r165015961)

**Review by grypez — APPROVED** [review]
LGTM
